src/Shared/FileMatcher.cs

```diff
@@ -1491,7 +1491,6 @@ private static int LastIndexOfDirectoryOrRecursiveSequence(string str, int start
             out bool isLegalFileSpec,
             FixupParts fixupParts = null)
         {
-            isLegalFileSpec = true;
```

Comment:
Did you check all the writes to `isLegalFileSpec` and make sure they do set it to true, and not just to false?

Reply:
Roslyn told me that this assignment is redundant and can be deleted. Roslyn is correct, on all filepaths the true value was overwritten.